azure/scope/managedmachinepool.go

```diff
@@ -281,6 +288,14 @@ func (s *ManagedMachinePoolScope) PatchCAPIMachinePoolObject(ctx context.Context
 	)
 }
 
+// PatchControlPlaneObject persists the AzureManagedControlPlane configuration and status.
+func (s *ManagedMachinePoolScope) PatchControlPlaneObject(ctx context.Context) error {
```

Comment:
I was wondering why you had to do this and then I saw https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2665/files#diff-5607bc7afbe8f18686e0e62796b8da6371e999006ddb598c5034dd7ef7040a47R234... I think the bug is actually that we're setting/getting the future on the wrong object above no? Probably a bad copy paste :(